pmd-core/src/main/java/net/sourceforge/pmd/lang/metrics/AbstractMetricsFacade.java

```diff
@@ -68,25 +68,24 @@ public double computeForType(MetricKey<T> key, T node, MetricVersion version) {
      *
      * @param key     The key identifying the metric to be computed
      * @param node    The node on which to compute the metric
-     * @param version The version of the metric
+     * @param options The options of the metric
      *
-     * @return The value of the metric, or {@code Double.NaN} if the value couln't be computed
+     * @return The value of the metric, or {@code Double.NaN} if the value couldn't be computed
      */
-    public double computeForOperation(MetricKey<O> key, O node,
-                                      MetricVersion version) {
+    public double computeForOperation(MetricKey<O> key, O node, MetricOptions options) {
 
         Objects.requireNonNull(key, "The metric key must not be null");
 
         if (!key.supports(node)) {
             return Double.NaN;
         }
 
-        MetricVersion safeVersion = (version == null) ? Version.STANDARD : version;
+        MetricOptions safeOptions = options == null ? MetricOptions.emptyOptions() : options;
```

Comment:
Hm.. Would it make sense to require the options to always be non-null? Then we could spare us the safeOptions.
This however would be slight API change - but would be consistent, since all parameters of this method must then be non-null....